infra/src/main/kotlin/fi/fta/geoviite/infra/geocoding/Geocoding.kt

```diff
@@ -662,12 +667,24 @@ fun getProjectedAddressPoint(projection: ProjectionLine, alignment: IAlignment):
     }
 }
 
+private data class AddressPointWalkResult(val addressPoints: List<AddressPoint?>, val alignmentWalkFinished: Boolean)
+
+private data class AlignmentPointInterval(val start: AlignmentPoint, val end: AlignmentPoint) {
+    fun interpolateAlignmentPointAtPortion(proportion: Double): AlignmentPoint =
+        interpolateToAlignmentPoint(start, end, proportion)
+
+    val referenceDirection: Double
+        get() = directionBetweenPoints(start, end)
+}
+
 private fun getProjectedAddressPoints(
     projectionLines: List<ProjectionLine>,
     alignment: IAlignment,
-): List<AddressPoint?> {
-    val walk = AlignmentWalk(getPolyLineEdges(alignment))
-    return projectionLines.map(walk::stepWith)
+): AddressPointWalkResult {
+    val alignmentPoints = alignment.allAlignmentPoints.zipWithNext(::AlignmentPointInterval).toList()
```

Comment:
GVT-3090 korjauksen tärkeä osa: Ongelma oli, että getPolyLineEdges toteuttaa tuon pituusmittauslinjoille tarkoitetun logiikan, että generoidun segmentin kulmaa ei oteta huomioon, vaan että siinä käytetään sitä edeltävää/sen jälkeen tulevaa kulmaa; tämän takia sitten AlignmentWalk#stepWith()in sisällä isEdgeAligned saa väärän arvon, koska sen tarkoitushan on katsoa juuri tämän yhden edgen kulmaa, mutta eipä katsottukaan, vaan katsottiin tätä "korjattua" kulmaa.

Kaikkien alignment-pisteiden haku maksaa suunnilleen saman paljon kuin getPolyLineEdges, eikä juuri näy profiileissa.